src/search3_httpd.erl

```diff
@@ -59,6 +59,52 @@ handle_search_req(#httpd{path_parts=[_, _, _, _, _]}=Req, _Db, _DDoc,
 handle_search_req(Req, _Db, _DDoc, _RetryCount, _RetryPause) ->
     send_error(Req, {bad_request, "path not recognized"}).
 
+handle_analyze_req(#httpd{method='GET'}=Req) ->
+    Analyzer = couch_httpd:qs_value(Req, "analyzer"),
+    Text = couch_httpd:qs_value(Req, "text"),
+    analyze(Req, Analyzer, Text);
+handle_analyze_req(#httpd{method='POST'}=Req) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    {Fields} = chttpd:json_body_obj(Req),
+    Analyzer = couch_util:get_value(<<"analyzer">>, Fields),
+    Text = couch_util:get_value(<<"text">>, Fields),
+    analyze(Req, Analyzer, Text);
+handle_analyze_req(Req) ->
+    send_method_not_allowed(Req, "GET,POST").
+
+analyze(Req, Analyzer, Text) ->
+    case Analyzer of
+        undefined ->
+            throw({bad_request, "analyzer parameter is mandatory"});
+        _ when is_list(Analyzer) ->
+            ok;
+        _ when is_binary(Analyzer) ->
+            ok;
+        {[_|_]} ->
+            ok;
+        _ ->
+            throw({bad_request, "analyzer parameter must be a string or an object"})
+    end,
+    case Text of
+        undefined ->
+            throw({bad_request, "text parameter is mandatory"});
+        _ when is_list(Text) ->
```

Comment:
same comment as above, but this shows the danger of this pattern. you're referring to a previously defined variable further up the function, easy to get wrong. Perhaps a few "is_string" and "is_string_or_object" functions could be pulled out for further clarity.